simulation/pathfinder/main.cc

```diff
@@ -1,19 +1,139 @@
+                    path.push_back(F_D);
+            /*though same change in the position,
+            due to robots direction (facing up or down) 
+            it can be the left turn or right turn*/
+
+            /*when robot is facing up*/
+            else if(init.row == (last.row+1) && path.back() == F_U){
+                    path.push_back(cc90);
+                    path.push_back(F_D);
+                    current_status = H;
+                    }
+            else if(init.row == (last.row-1) && path.back() == F_U){
+                    path.push_back(cw90);
+                    path.push_back(F_U);
+                    current_status = H;
+                    }
+            /*when robot is facing up*/
+
+            /*when robot is facing down*/
+            else if(init.row == (last.row+1) && path.back() == F_D){
+                    path.push_back(cw90);
+                    path.push_back(F_D);
+                    current_status = H;
+                    }
+            else if(init.row == (last.row-1) && path.back() == F_D){
+                    path.push_back(cc90);
+                    path.push_back(F_U);
+                    current_status = H;
+                    }
+            /*when robot is facing down*/
+        } else if(current_status == H)  {
+            if(init.row == (last.row-1))
+                    path.push_back(F_U);
+            else if(init.row == (last.row+1) )
+                    path.push_back(F_D);
+            /*though same change in the position,
+            due to robots direction (facing up or down) 
+            it can be the left turn or right turn*/
+
+            /*when robot is facing up*/
+            else if(init.col == (last.col+1) && path.back() == F_U){
+                    path.push_back(cw90);
+                    path.push_back(F_D);
+                    current_status = V;
+                    }
+            else if(init.col == (last.col-1) && path.back() == F_U){
+                    path.push_back(cc90);
+                    path.push_back(F_U);
+                    current_status = V;
+                    printf("%d. %d -> %d,%d   /  %c \n",init.row,init.col, last.row,last.col,path.back());
+                    }
+            /*when robot is facing up*/
+
+            /*when robot is facing down*/
+            else if(init.col == (last.col+1) && path.back() == F_D){
+                    path.push_back(cc90);
+                    path.push_back(F_D);
+                    current_status = V;
+                    }
+            else if(init.col == (last.col-1) && path.back() == F_D){
+                    path.push_back(cw90);
+                    path.push_back(F_U);
+                    current_status = V;
+                    }
+            /*when robot is facing down*/
+                           
+                   
+        }
+        
+    }
+
+    for(int i=0; path.front() ; i++){
+            
+    }
+
+    for(int i=0; path.front() ; i++)    {
```

Comment:
According to [a C++ documentation on `deque::front()`](http://www.cplusplus.com/reference/deque/deque/front/):
> Calling this function on an empty container causes undefined behavior.

In programing, "undefined behavior" means it may cause a catastrophic failure. NEVER rely on these behavior. Please read a documentation before using it. 

There is a correct way to do it:
```C++
for(int i=0; !path.empty() ; i++) {
    do_stuff
}
```